test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13930,16 +13932,23 @@ def create_script(self, body=None, asuser=None, hostname=None):
                     body[i] = " ".join(line_arr)
             body = '\n'.join(body)
 
+        if self.platform == 'shasta' and self.username:
+            user = PbsUser.get_user(self.username)
+            if user.host:
+                hostname = user.host
+                shasta_set_host = True
+
         self.script_body = body
         if self.du is None:
             self.du = DshUtils()
         # First create the temporary file as current user and only change
         # its mode once the current user has written to it
         fn = self.du.create_temp_file(hostname, prefix='PtlPbsJobScript',
                                       asuser=asuser, body=body)
-
         self.du.chmod(hostname, fn, mode=0755)
-        if not self.du.is_localhost(hostname):
+        # on shasta, if we have already set the hostname,
+        # it does not need to be copied again
```

Comment:
Could you add a why in here?  Have you already copied it?  Maybe say that and where in the code that copied it.

Reply:
I'm not even sure why this is in the code, create_temp_file should create the job script on the host specified by hostname. We then chmod the file on that host. Then we do another copy? It seems wrong, I'll remove it.